src/basic/fs-util.c

```diff
@@ -1083,3 +1083,42 @@ int open_mkdir_at(int dirfd, const char *path, int flags, mode_t mode) {
 
         return TAKE_FD(fd);
 }
+
+int openat_report_new(int dirfd, const char *pathname, int flags, mode_t mode, bool *ret_newly_created) {
+        unsigned attempts = 7;
+
+        /* Just like openat(), but adds one thing: optionally returns whether we created the file anew or if
+         * it already existed before. This is only relevant of O_CREAT is set without O_EXCL, and thus will
+         * shortcut to openat() otherwise */
+
+        if (!FLAGS_SET(flags, O_CREAT) || FLAGS_SET(flags, O_EXCL) || !ret_newly_created)
```

Comment:
Since the main (only ?) reason to use this function is to figure out whether the file has been created or not, I would make the presence of `ret_newly_created` mandatory.

Reply:
Also whenever `ret_newly_created` is passed, we should initialize it. I think the caller can expect that regardless of the value of the flags.